src/main/java/marites/Marites.java

```diff
@@ -0,0 +1,312 @@
+    private static final String SET_TASK_STATUS_FORMAT_STRING = "%s%n    %s\n";
+
+    private static final Scanner SCANNER = new Scanner(System.in);
+    public static final String SAVE_FILENAME = "tasklist.ser";
+    public static final String READ_TASK_LIST_ERROR_MESSAGE = "WARNING: Error when reading saved task list; initializing with empty list";
+    public static final String SAVE_TASK_LIST_ERROR_MESSAGE = "WARNING: Error occurred while saving task list";
+
+    /** The task list being tracked by marites.Marites. */
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+
+
+    public static void main(String[] args) {
+        initializeTaskList();
+        printIntroduction();
+        while (true) {
+            String userInput = SCANNER.nextLine();
+            String commandFeedback = processUserCommand(userInput);
+            showFeedback(commandFeedback);
+        }
+    }
+
+    private static void initializeTaskList() {
+        File saveFile = new File(SAVE_FILENAME);
+        if (!saveFile.exists()) {
+            tasks = new ArrayList<>();
+            return;
+        }
+        try {
+            FileInputStream fileIn = new FileInputStream(SAVE_FILENAME);
+            ObjectInputStream objIn = new ObjectInputStream(fileIn);
+            tasks = (ArrayList<Task>)objIn.readObject();
+        } catch (IOException | ClassNotFoundException | ClassCastException e) {
+            System.out.println(READ_TASK_LIST_ERROR_MESSAGE);
+            tasks = new ArrayList<>();
+        }
+    }
+
+    private static void saveTaskList() {
+        try {
+            File saveFile = new File(SAVE_FILENAME);
+            saveFile.createNewFile();
+            FileOutputStream fileOut = new FileOutputStream(SAVE_FILENAME);
+            ObjectOutputStream objOut = new ObjectOutputStream(fileOut);
+            objOut.writeObject(tasks);
+        } catch (IOException e) {
+            System.out.println(SAVE_TASK_LIST_ERROR_MESSAGE);
+        }
+    }
+
+    /* OUTPUT */
+
+    /**
+     * Prints a separator string.
+     */
+    private static void printSeparator() {
+        System.out.println("========================================");
+    }
+
+    /**
+     * Prints an introductory message.
+     */
+    private static void printIntroduction() {
+        System.out.println(INTRO_MESSAGE);
+        printSeparator();
+    }
+    /**
+     * Prints an exit message.
+     */
+    private static void printExitMessage() {
+        System.out.println(EXIT_MESSAGE);
+        printSeparator();
+    }
+
+    /**
+     * Prints a feedback string from an executed command.
+     * @param feedback The feedback string to execute.
+     */
+    private static void showFeedback(String feedback) {
+        printSeparator();
+        System.out.print(feedback);
+        printSeparator();
+    }
+
+    /**
+     * Divides a line of user input into the command type, and the command.
+     * @param userInput The user's input
```

Comment:
**Coding standard violation:** Separate these two lines with an empty line

For your reference: https://se-education.org/guides/conventions/java/index.html#comments